integrations/gcp/gcp_core/gcp_client.py

```diff
@@ -0,0 +1,109 @@
+from collections.abc import MutableSequence
+from typing import Any, Optional, TypeVar
+import proto  # type: ignore
+from google.api_core.exceptions import PermissionDenied, NotFound
+from google.cloud.asset_v1 import (
+    AssetServiceAsyncClient,
+    ContentType,
+)
+from google.cloud.asset_v1.services.asset_service import pagers
+from loguru import logger
+from google.pubsub_v1.services.publisher import PublisherAsyncClient
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE, RAW_ITEM
+
+RESOURCE_METADATA_CONTENT_TYPE = ContentType(1)
+GCP_PUBSUB_TOPIC_ASSET_KIND = "pubsub.googleapis.com/Topic"
+TOPIC_PROJECT_FIELD = "__project_id"
+GCP_RESOURCE_MANAGER_PROJECT_KIND = "cloudresourcemanager.googleapis.com/Project"
```

Comment:
can u create an enum for known kinds and then move the existing enum for specifically supported kind list?

Reply:
For all known kinds? or only to those i have to refer to in the code